daemon/clone-mgr.c

```diff
@@ -1150,25 +1162,41 @@ seaf_clone_manager_add_task (SeafCloneManager *mgr,
     char *worktree;
     char *ret;
     gboolean sync_wt_name = FALSE;
+    char *repo_salt;
 
     if (!seaf->started) {
         seaf_message ("System not started, skip adding clone task.\n");
-        return NULL;
+        goto out;
     }
 
 #ifdef USE_GPL_CRYPTO
     if (repo_version == 0 || (passwd && enc_version < 2)) {
         seaf_warning ("Don't support syncing old version libraries.\n");
         g_set_error (error, SEAFILE_DOMAIN, SEAF_ERR_BAD_ARGS,
                      "Don't support syncing old version libraries");
-        return NULL;
+        goto out;
     }
 #endif
 
-    if (passwd &&
-        !check_encryption_args (magic, enc_version, random_key, error))
-        return NULL;
+    if (more_info) {
+        json_error_t jerror;
+        json_t *object = NULL;
```

Comment:
It's not necessary to set object to NULL, since json_loads will return NULL on failure.